include/libtorrent/piece_picker.hpp

```diff
@@ -850,6 +852,7 @@ namespace libtorrent {
 		// the number of pieces we have that also are filtered
 		int m_num_have_filtered = 0;
 
+		piece_index_t m_sequential_start {0};
```

Comment:
if `m_cursor` and `m_reverse_cursor` are only used for sequential download, I think it makes sense to use those instead of adding a new field. Their semantics would have to be altered a bit though, the comments wouldn't necessarily be true

Reply:
@arvidn Okay. This is a difficult question for me. It has to be tested. Do you want to assign m_cursor?

Reply:
@arvidn Many factors affect m_cursor. Are you sure there won't be a problem with asynchronous assignment? Perhaps a problem with concurrency?

Reply:
no, I'm not sure. but iirc, the only purpose that member exists is for implementing sequential download. I would expect it make for simpler code, overall, to tweak the semantics of this rather than adding another member

Reply:
@arvidn I'm going to test it now and let you know the results.

Reply:
@arvidn I just checked it out.
If I remove the m_sequential_start variable and use m_cursor instead, the data download fading starts to appear. Only the data to which deadline is assigned gets downloaded.